pygmt/tests/test_legend.py

```diff
@@ -2,12 +2,20 @@
 Tests for legend.
 """
 import pytest
-from pygmt import Figure
+from packaging.version import Version
+from pygmt import Figure, clib
 from pygmt.exceptions import GMTInvalidInput
 from pygmt.helpers import GMTTempFile
 
+with clib.Session() as _lib:
+    gmt_version = Version(_lib.info["version"])
+
 
 @pytest.mark.mpl_image_compare
+@pytest.mark.xfail(
+    condition=gmt_version > Version("6.3.0"),
+    reason="Defaults updated in https://github.com/GenericMappingTools/gmt/pull/6165",
+)
```

Comment:
Probably doesn't matter, but wondering if it's better to have the xfail mark be on top of mpl_image_compare. Maybe trigger the GMT Dev Tests CI to be sure.
```suggestion
@pytest.mark.xfail(
    condition=gmt_version > Version("6.3.0"),
    reason="Defaults updated in https://github.com/GenericMappingTools/gmt/pull/6165",
)
@pytest.mark.mpl_image_compare
```

Reply:
I think it doesn't matter because pygmt/tests/test_plot3d.py::test_plot3d_matrix_color has it on top while pygmt/tests/test_plot.py::test_plot_matrix has it on the bottom, so we could just pick one to be consistent.

Reply:
Ok, let's just keep it as it is now at the bottom.